main.go

```diff
@@ -1,5 +1,73 @@
 package main
 
-func main() {
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"sort"
+)
+
+type HumanDecoding interface {
```

Comment:
https://go.dev/doc/effective_go#interface-names
Interface names should be ended with 'er'